drivers/adc/adc_stm32.c

```diff
@@ -1347,6 +1348,117 @@ static int adc_stm32_channel_setup(const struct device *dev,
 	return 0;
 }
 
+#if defined(CONFIG_SOC_SERIES_STM32C0X) || defined(CONFIG_SOC_SERIES_STM32G0X) ||                  \
+	defined(CONFIG_SOC_SERIES_STM32L0X) ||                                                     \
+	(defined(CONFIG_SOC_SERIES_STM32WBX) && defined(ADC_SUPPORT_2_5_MSPS)) ||                  \
+	defined(CONFIG_SOC_SERIES_STM32WLX) || defined(CONFIG_SOC_SERIES_STM32H7X)
+static bool adc_stm32_is_clk_sync(const struct adc_stm32_cfg *config)
```

Comment:
I don't really like to have the same `#if defined` duplicated in two places.
I would prefer to have something like
```
#if defined(CONFIG_SOC_SERIES_STM32C0X) || \
	defined(CONFIG_SOC_SERIES_STM32G0X) || \
	defined(CONFIG_SOC_SERIES_STM32L0X) || \
	(defined(CONFIG_SOC_SERIES_STM32WBX) && defined(ADC_SUPPORT_2_5_MSPS)) || \
	defined(CONFIG_SOC_SERIES_STM32WLX)
	define HAS_INDIVIDUAL_CLOCK_MODE
#endif
```
And then use this define in the two places.
I'm not too sure about the define name though, don't hesitate to propose something else if you have ideas.
Note that keeping one SOC_SERIES per line is preferred for readability and adding potential future series.